web/bridge_types_controller.go

```diff
@@ -19,11 +22,23 @@ func (btc *BridgeTypesController) Create(c *gin.Context) {
 		c.JSON(500, gin.H{
 			"errors": []string{err.Error()},
 		})
-	} else if err = services.ValidateAdaptor(bt, btc.App.GetStore()); err != nil {
+	} else {
+		btc.SaveAdapter(c, bt)
+	}
+}
+
+// SaveAdapter saves an adapter if it is valid
+func (btc *BridgeTypesController) SaveAdapter(c *gin.Context, bt *models.BridgeType) {
+	store := btc.App.GetStore()
+
```

Comment:
I think this is the right idea, but in the wrong place. If we ever need to `SaveAdapter` in another controller or somewhere else, we'll run into the same issue and have to duplicate the code. Let's make this easier to reuse.

We need access to both the `services` package and the `store` package, so let's put it in `application.go` below `AddJob`. Let's also rename it to `AddAdapter` for consistency.

below services/application.go:87
https://github.com/smartcontractkit/chainlink/blob/master/services/application.go#L87

If you think this is better or there's another better way, lmk.

Reply:
Makes sense I'll update code tonight